test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller_phases.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	matchingMachines := machinesMatchingInfrastructureSpec(ctx, machines, machinePool, dockerMachinePool)
+	for _, machine := range matchingMachines {
+		if err := r.Client.Get(ctx, client.ObjectKey{Namespace: machinePool.Namespace, Name: machine.Name()}, &infrav1.DockerMachine{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				if err := r.createDockerMachine(ctx, machine.Name(), cluster, machinePool, dockerMachinePool); err != nil {
+					return errors.Wrap(err, "failed to create a new docker machine")
+				}
+			} else {
+				return errors.Wrap(err, "failed to get docker machine")
+			}
+		} else {
+			log.Info("DockerMachine already exists, nothing to do", "name", machine.Name())
+		}
+	}
+
+	return nil
+}
+
+// createDockerMachine creates a DockerMachine to represent a Docker container in a DockerMachinePool.
+// These DockerMachines have the clusterv1.ClusterNameLabel and clusterv1.MachinePoolNameLabel to support MachinePool Machines.
+func (r *DockerMachinePoolReconciler) createDockerMachine(ctx context.Context, name string, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	labels := map[string]string{
+		clusterv1.ClusterNameLabel:     cluster.Name,
+		clusterv1.MachinePoolNameLabel: format.MustFormatValue(machinePool.Name),
+	}
+	dockerMachine := &infrav1.DockerMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   dockerMachinePool.Namespace,
+			Name:        name,
+			Labels:      labels,
+			Annotations: make(map[string]string),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: dockerMachinePool.APIVersion,
+					Kind:       dockerMachinePool.Kind,
+					Name:       dockerMachinePool.Name,
+					UID:        dockerMachinePool.UID,
+				},
+				// Note: Since the MachinePool controller has not created its owner Machine yet, we want to set the DockerMachinePool as the owner so it's not orphaned.
+			},
+		},
+		Spec: infrav1.DockerMachineSpec{
+			CustomImage:   dockerMachinePool.Spec.Template.CustomImage,
+			PreLoadImages: dockerMachinePool.Spec.Template.PreLoadImages,
+		},
+	}
+
+	log.V(2).Info("Creating DockerMachine", "dockerMachine", dockerMachine.Name)
+
+	if err := r.Client.Create(ctx, dockerMachine); err != nil {
+		return errors.Wrap(err, "failed to create dockerMachine")
+	}
+
+	return nil
+}
+
+// deleteExtraDockerMachines deletes the DockerMachines needed to reach the MachinePool's desired replica count in a scale down as well as any referencing Docker containers that do not match the infrastructure spec.
+func (r *DockerMachinePoolReconciler) deleteExtraDockerMachines(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.V(2).Info("Deleting extra machines if needed", "dockerMachinePool", dockerMachinePool.Name, "namespace", dockerMachinePool.Namespace)
```

Comment:
Is this part of docker machine pool machine core logic?

Reply:
Yes, since the MachinePool controller does not delete Machines, it is being done here.